From: Thomas Renninger <trenn@suse.de>
Subject: Only use 32 bit addresses if they have a valid length
References: bug#581644
Patch-Mainline: not yet

Also not sure whether it will help, but it's a fix.

Please remove this patch again after a while also if it's not
mainline.

---
 drivers/acpi/acpica/tbfadt.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/acpi/acpica/tbfadt.c
+++ b/drivers/acpi/acpica/tbfadt.c
@@ -550,11 +550,12 @@ static void acpi_tb_validate_fadt(void)
 			    (!address64->address && length)) {
 				ACPI_WARNING((AE_INFO,
 					      "Optional field %s has zero address or length: "
-					      "0x%8.8X%8.8X/0x%X",
+					      "0x%8.8X%8.8X/0x%X - not using it",
 					      name,
 					      ACPI_FORMAT_UINT64(address64->
 								 address),
 					      length));
+				address64->address = 0;
 			}
 		}
 	}
